aspnetcore/test/integration-tests/samples/2.x/IntegrationTestsSample/tests/RazorPagesProject.Tests/IntegrationTests/AuthTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using AngleSharp.Dom.Html;
+using Xunit;
+using RazorPagesProject.Services;
+using RazorPagesProject.Tests.Helpers;
+
+namespace RazorPagesProject.Tests
+{
+    public class AuthTests : 
+        IClassFixture<CustomWebApplicationFactory<RazorPagesProject.Startup>>
+    {
+        private readonly CustomWebApplicationFactory<RazorPagesProject.Startup> 
+            _factory;
+
+        public AuthTests(
+            CustomWebApplicationFactory<RazorPagesProject.Startup> factory)
+        {
+            _factory = factory;
+        }
+
+        #region snippet1
+        [Fact]
+        public async Task Get_GithubProfilePageCanGetAGithubUser()
+        {
+            // Arrange
+            void ConfigureTestServices(IServiceCollection services) =>
```

Comment:
What is this `void` doing here?

Reply:
Just like `Startup.ConfigureServices` in an app doesn't return when it's called by the runtime and configures services for the app's container, this pattern works the same way. This sets up services that pertain to this particular test. `ConfigureTestServices` is passed to `builder.ConfigureTestServices`. In this test, the service collection needs `TestGithubClient` registered for `IGithubClient`. @javiercn will correct my explanation if I'm off base there.